apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns.php

```diff
@@ -45,112 +62,116 @@ public static function get_instance() {
 	/**
 	 * Register FSE block patterns and categories.
 	 */
-	public function register_patterns() {
-		// Register Block Pattern Categories.
+	private function register_patterns() {
+		// Remove core pattern categories, categories added later that match
+		// will then remain in alphabetical order.
 		if ( class_exists( 'WP_Block_Pattern_Categories_Registry' ) ) {
-			register_block_pattern_category( 'blog', array( 'label' => _x( 'Blog', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'call-to-action', array( 'label' => _x( 'Call to Action', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'contact', array( 'label' => _x( 'Contact', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'images', array( 'label' => _x( 'Images', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'list', array( 'label' => _x( 'List', 'Block pattern category', 'full-site-editing' ) ) );
-
-			// The 'Two Columns of Text' pattern is in the 'columns' and 'text' categories.
-			// Removing 'columns' so it doesn't appear as a category with only a single item.
+			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'buttons' ) ) {
+				unregister_block_pattern_category( 'buttons' );
+			}
 			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'columns' ) ) {
 				unregister_block_pattern_category( 'columns' );
 			}
+			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'gallery' ) ) {
+				unregister_block_pattern_category( 'gallery' );
+			}
+			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'header' ) ) {
+				unregister_block_pattern_category( 'header' );
+			}
+			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'text' ) ) {
+				unregister_block_pattern_category( 'text' );
+			}
```

Comment:
Do we effectively want to first look up ALL core blocks and then unregister them? Can we get a list from somewhere and just loop through them all?

Reply:
We actually don't need this code at all. Since we're unregistering core blocks in the next block of code, the core categories will not show up because they contain no blocks. Removing this code also means we won't trample on third party blocks that use these categories.